packages/flutter/lib/src/painting/decoration.dart

```diff
@@ -180,13 +180,12 @@ abstract class BoxPainter {
   ///
   /// Resources might not start to load until after [paint] has been called,
   /// because they might depend on the configuration.
-  VoidCallback get onChanged => _onChanged;
-  VoidCallback _onChanged;
+  final VoidCallback onChanged;
 
-  /// Discard any resources being held by the object. This also guarantees that
-  /// the [onChanged] callback will not be called again.
+  /// Discard any resources being held by the object.
```

Comment:
Can the `_onChanged` callback hold onto any resources that may no longer be GC'd after this change?

Reply:
The assumption is that if you're calling dispose, you'll not be holding onto this object in the first place, so the callback will be GC'ed along with the object itself. We need the dispose just to make sure that we stop listening (which _would_ prevent GC since the image itself is held by the cache).